src/utils/getNumberOfCalendarMonthWeeks.js

```diff
@@ -9,7 +9,7 @@ export default function getNumberOfCalendarMonthWeeks(
   month,
   firstDayOfWeek = moment.localeData().firstDayOfWeek(),
 ) {
-  const firstDayOfMonth = month.clone().startOf('month');
+  const firstDayOfMonth = month.clone().startOf('month').hour(12);
```

Comment:
i assume that `getNumberOfCalendarMonthWeeks` has a bug in some timezones if the hour is set to 0 - can we have a unit test that triggers that bug, such that this line's change fixes it?

Reply:
To be honest, getNumberOfCalendarMonthWeeks() is not affected by the issue, I added "hour(12)" as unified approach to get firstDayOfMonth. Probably it's a good idea to have special util methods like :
```js
const INITIAL_HOUR = 12;

getFirstDayOfMonth = (month) => month.clone().startOf('month').hour(INITIAL_HOUR);
getFirstDayOfFirstWeekOfMonth = (month) => getFirstDayOfMonth(month).startOf('week').hour(INITIAL_HOUR);
```

Reply:
Is there no timezone/date combination where the first day of the month, or the first day of the first week of a month, doesn't have a midnight?

Reply:
Midnight is 00:00, but I set hour(12) that is a noon to mitigate summer/winter time changes, so noon is a good choice for date calculations. Or you mean something other, sorry?

Reply:
Right - I’m envisioning a bug caused by midnight that noon fixes.

Reply:
I did quick research and didn't find any other issues related to midnight, but react-dates codebase is large and no warranty here without full code audit.

Reply:
What I mean is, 2013.10.20 was the start of the week in Brazil where midnight doesn't exist, but it's not the start of the month or the first day of the week when the week ends in a different month. Maybe there's another date in a specific timezone where either of those is true?

Reply:
> where midnight doesn't exist
I don't see that lack of midnight causes any error:
```js
const moment = require('moment-timezone');
moment.tz("2013-10-20 00:00:00", "Brazil/East").toString();
```
Output:
```Sun Oct 20 2013 01:00:00 GMT-0200```
Moment handles this flawlessly as I can see(+1 hour added).

Thank you for merging!